### PR TITLE
Org deletion job setup

### PR DESCRIPTION
**Pull request by jjidt**
- sets up endpoints for canceled org deletion
- moves admin auth to header `Authorization: Admin <adminsecret>`
- sets up kubernetes cronjob

### CODE REVIEW

File: kubernetes/jobs/configMaps.yaml [hunk 1]
```diff
@@ -9,4 +9,11 @@ items:
   metadata:
     name: audit-log-backup-config
     namespace: default
+- apiVersion: v1
+  data:
+    delete.api_url: 'https://api.portway.app'
```
Comment (dheniges) [inline kubernetes/jobs/configMaps.yaml]: I would re-use the existing config map that already has this url. I think it's in the web config map, which arguably should be renamed/moved into its own config map, but I wouldn't worry about that for now. So you can drop the changes to this file

File: kubernetes/jobs/deletedOrgCleanup.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          restartPolicy: OnFailure
+          containers:
+          - name: deleted-org-cleanup
+            image: bonkeybong/portway_deleted_org_cleanup
+            env:
+            - name: ADMIN_API_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: deleted-org-cleanup
+                  key: admin_api_key
```
Comment (dheniges) [inline kubernetes/jobs/deletedOrgCleanup.yaml]: We already have the admin key used in a secret, which you can see in the api deployment. Re-use that secret here. (note you can keep the `ADMIN_API_KEY` env var name and define it as the `admin_secret_key`)

File: kubernetes/jobs/deletedOrgCleanup.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            - name: LOG_ID
+              valueFrom:
+                secretKeyRef:
+                  name: deleted-org-cleanup
+                  key: log_id
+            - name: API_URL
+              valueFrom:
+                configMapKeyRef:
+                  name: deleted-org-cleanup-config
+                  key: delete.api_url
```
Comment (dheniges) [inline kubernetes/jobs/deletedOrgCleanup.yaml]: this is where you can change the definition to pickup the web config map's public api url key

File: kubernetes/jobs/deletedOrgCleanup.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+          containers:
+          - name: deleted-org-cleanup
+            image: bonkeybong/portway_deleted_org_cleanup
+            env:
+            - name: ADMIN_API_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: deleted-org-cleanup
+                  key: admin_api_key
+            - name: LOG_ID
```
Comment (dheniges) [inline kubernetes/jobs/deletedOrgCleanup.yaml]: doesn't look like LOG_ID is used in the job, you can delete this env var

**Review by dheniges — CHANGES_REQUESTED** [review]
Feedback on the kube config

**Comment by jjidt** [comment]
updated

**Review by dheniges — APPROVED** [review]
Gets canceled orgs, can't delete canceled ones less than 30 days old, or non-canceled orgs  👍

**Pull request merged**